benchmark.py

```diff
@@ -0,0 +1,307 @@
+            landsat.index,
+            size=args.patch_size,
+            batch_size=args.batch_size,
+            length=length,
+            roi=roi,
+        ),
+    ]
+
+    results_rows = []
+    for sampler in samplers:
+        if args.verbose:
+            print(f"\n{sampler.__class__.__name__}:")
+
+        if isinstance(sampler, RandomBatchGeoSampler):
+            dataloader = DataLoader(
+                dataset,
+                batch_sampler=sampler,  # type: ignore[arg-type]
+                num_workers=args.num_workers,
+            )
+        else:
+            dataloader = DataLoader(
+                dataset,
+                batch_size=args.batch_size,
+                sampler=sampler,  # type: ignore[arg-type]
+                num_workers=args.num_workers,
+            )
+
+        tic = time.time()
+        num_total_patches = 0
+        for i, batch in enumerate(dataloader):
+            num_total_patches += args.batch_size
+            # This is to stop the GridGeoSampler from enumerating everything
+            if i == num_batches - 1:
+                break
+        toc = time.time()
+        duration = toc - tic
+
+        if args.verbose:
+            print(f"  duration: {duration:.3f} sec")
+            print(f"  count: {num_total_patches} patches")
+            print(f"  rate: {num_total_patches / duration:.3f} patches/sec")
+
+        if args.cache:
+            if args.verbose:
+                print(landsat._cached_load_warp_file.cache_info())
+
+            # Clear cache for fair comparison between samplers
+            # Both `landsat` and `cdl` share the same cache
+            landsat._cached_load_warp_file.cache_clear()
+
+        results_rows.append(
+            {
+                "cached": args.cache,
+                "seed": args.seed,
+                "duration": duration,
+                "count": num_total_patches,
+                "rate": num_total_patches / duration,
+                "sampler": sampler.__class__.__name__,
+                "batch_size": args.batch_size,
+                "num_workers": args.num_workers,
+            }
+        )
+
+    # Benchmark model
+    model = resnet18()
+    # Change number of input channels to match Landsat
+    model.conv1 = nn.Conv2d(  # type: ignore[attr-defined]
+        len(bands), 64, kernel_size=7, stride=2, padding=3, bias=False
+    )
+
+    criterion = nn.CrossEntropyLoss()  # type: ignore[attr-defined]
+    params = model.parameters()
+    optimizer = optim.SGD(params, lr=0.0001)
+
+    device = torch.device(  # type: ignore[attr-defined]
+        "cuda" if torch.cuda.is_available() else "cpu"
+    )
+    model = model.to(device)
+
+    tic = time.time()
+    num_total_patches = 0
+    for _ in range(num_batches):
+        num_total_patches += args.batch_size
+        x = torch.rand(args.batch_size, len(bands), args.patch_size, args.patch_size)
+        # y = torch.randint(0, 256, (args.batch_size, args.patch_size, args.patch_size))
```

Comment:
This line is left over from when I was thinking about benchmarking a segmentation model instead of ResNet. I think segmentation is one of the more common tasks in remote sensing, and the models are more complex and therefore slower. If we want to have a slower model for comparison with our data loading rates, it might be good to use something like Mask R-CNN instead.